packages/subscription-controller/src/SubscriptionController.ts

```diff
@@ -222,6 +249,124 @@ export class SubscriptionController extends BaseController<
+      throw new Error('Product price not found');
+    }
+
+    const price = product.prices.find((p) => p.interval === request.interval);
+    if (!price) {
+      throw new Error('Price not found');
+    }
+
+    const chainsPaymentInfo = pricing.paymentMethods.find(
+      (t) => t.type === PaymentType.byCrypto,
+    );
+    if (!chainsPaymentInfo) {
+      throw new Error('Chains payment info not found');
+    }
+    const chainPaymentInfo = chainsPaymentInfo.chains?.find(
+      (t) => t.chainId === request.chainId,
+    );
+    if (!chainPaymentInfo) {
+      throw new Error('Invalid chain id');
+    }
+    const tokenPaymentInfo = chainPaymentInfo.tokens.find(
+      (t) => t.address === request.paymentTokenAddress,
+    );
+    if (!tokenPaymentInfo) {
+      throw new Error('Invalid token address');
+    }
+
+    const tokenApproveAmount = this.#getTokenApproveAmount(
+      price,
+      tokenPaymentInfo,
+    );
+
+    return {
+      approveAmount: tokenApproveAmount.toString(),
+      spenderAddress: chainPaymentInfo.paymentAddress,
+      paymentTokenAddress: request.paymentTokenAddress,
+      chainId: request.chainId,
+    };
+  }
+
+  /**
+   * Calculate total subscription price amount from price info
+   * e.g: $8 per month * 12 months min billing cycles = $96
+   *
+   * @param price - The price info
+   * @returns The price amount
+   */
+  #getSubscriptionPriceAmount(price: ProductPrice) {
+    // no need to use BigInt since max unitDecimals are always 2 for price
+    const amount =
+      (price.unitAmount / 10 ** price.unitDecimals) * price.minBillingCycles;
+    return amount;
+  }
+
+  /**
+   * Calculate token approve amount from price info
+   *
+   * @param price - The price info
+   * @param tokenPaymentInfo - The token price info
+   * @returns The token approve amount
+   */
+  #getTokenApproveAmount(
+    price: ProductPrice,
+    tokenPaymentInfo: TokenPaymentInfo,
+  ) {
+    const conversionRate =
+      tokenPaymentInfo.conversionRate[
+        price.currency as keyof typeof tokenPaymentInfo.conversionRate
+      ];
+    if (!conversionRate) {
+      throw new Error('Conversion rate not found');
+    }
+    // conversion rate is a float string e.g: "1.0"
+    // We need to handle float conversion rates with integer math for BigInt.
+    // We'll scale the conversion rate to an integer by multiplying by 10^4.
+    // conversionRate is in usd decimal. In most currencies, we only care about 2 decimals (cents)
+    // So, scale must be max of 10 ** 4 (most exchanges trade with max 4 decimals of usd)
+    // This allows us to avoid floating point math and keep precision.
+    const CONVERSION_RATE_SCALE = 10n ** 4n;
+    const conversionRateScaled = BigInt(
+      Number(conversionRate) * Number(CONVERSION_RATE_SCALE),
+    );
+    const priceAmount = this.#getSubscriptionPriceAmount(price);
+    const priceAmountScaled = BigInt(
+      priceAmount * Number(CONVERSION_RATE_SCALE),
+    );
+
+    const amount =
```

Comment:
it looks like we are redundantly multiply and dividing with CONVERSION_RATE_SCALE,  above code can be simplified as:

```
const SCALE = 10n ** 4n;
const conversionRateScaled = BigInt(Number(conversionRate) * Number(SCALE));
const priceAmount = this.#getSubscriptionPriceAmount(price);
const priceAmountScaled = BigInt(priceAmount * Number(SCALE));
const tokenDecimals = BigInt(tokenPaymentInfo.decimals);

const amount = (priceAmountScaled * (10n ** tokenDecimals)) / conversionRateScaled;

```

Reply:
I'm seeing
```
priceAmountScaled = BigInt(priceAmount * Number(CONVERSION_RATE_SCALE));
```
and then in then immediately in the next statement
```
... (priceAmountScaled / CONVERSION_RATE_SCALE) * ...
```
This just reverts the scaling!?